### PR TITLE
Use server_name instead of sitename for subid

### PR DESCRIPTION
**Pull request by marc1706**
A change in the server name is a lot less likely and admins will
therefore not receive a new subId unless they change hostnames.

### CODE REVIEW

**Comment by VSEphpbb** [comment]
Why don't we just add a migration that generates and stores a unique ID (or maybe create a hash of the sitename) in the config table. Then we can use that unique value plus the questionairre ID to generate the board's sub-id. It would only change then if the user purges VigLink and reinstalls again.

Or we could be cheeky and leave that config value behind, so it never changes. Or we could even have phpBB generate and store the value, and in either case just have VigLink use a Migration to create it if for some reason it doesn't exist. We can call the config ``viglink_sub_id``

**Comment by michaelcullum** [comment]
Because we want to get the server_name on .com, not an md5 hash.

**Comment by VSEphpbb** [comment]
@michaelcullum

File: acp/viglink_module.php [hunk 1]
```diff
@@ -111,12 +111,12 @@ public function main($id, $mode)
...
 
 		// Try to get convert account key from .com
-		$sub_id = md5(urlencode($config['sitename']) . $config['questionnaire_unique_id']);
+		$sub_id = md5($config['viglink_api_servername'] . $config['questionnaire_unique_id']);
 		$convert_account_link = $config->offsetGet('viglink_convert_account_url');
 
 		if (empty($convert_account_link) || strpos($config['viglink_convert_account_url'], 'subId=' . $sub_id) === false)
 		{
-			$convert_account_link = @file_get_contents('https://www.phpbb.com/viglink/convert?sitename=' . urlencode($config['sitename']) . '&amp;uuid=' . $config['questionnaire_unique_id'] . '&amp;key=' . $config['phpbb_viglink_api_key']);
+			$convert_account_link = @file_get_contents('https://www.phpbb.com/viglink/convert?domain=' . urlencode($config['server_name']) . '&amp;sitename=' . $config['viglink_api_servername'] . '&amp;uuid=' . $config['questionnaire_unique_id'] . '&amp;key=' . $config['phpbb_viglink_api_key']);
```
Comment (VSEphpbb) [inline acp/viglink_module.php]: So `sitename=' . $config['viglink_api_servername']`

But it's not a sitename anymore. It's a server name. Does this matter ??

Comment (marc1706) [inline acp/viglink_module.php]: Doesn't really matter. In the settings it's called `Domain` IIRC, so doesn't really matter. Just need to make sure the website expects this (which it does at the moment).

Comment (VSEphpbb) [inline acp/viglink_module.php]: maybe we shouldn't use 'sitename' as the url param anymore? maybe name it something else? 'api_servername'?

**Comment by marc1706** [comment]
@VSEphpbb renamed everything to site ID now

File: migrations/viglink_data.php [hunk 2]
```diff
@@ -29,14 +29,15 @@ public function update_data()
 	{
 		return array(
 			// Basic config options
-			array('config.add', array('viglink_enabled', 0)),
+			array('config.add', array('viglink_enabled', 1)),
```
Comment (VSEphpbb) [inline migrations/viglink_data.php]: Enabling it on install now?

Comment (DavidIQ) [inline migrations/viglink_data.php]: Yes.

Comment (DavidIQ) [inline migrations/viglink_data.php]: The extension itself was always installed and enabled. VigLink itself was not.

**Comment by VSEphpbb** [comment]
@michaelcullum is this PR good for you too?

**Pull request merged**